### PR TITLE
Add Aura experiment

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/488551667048375/1208893765398547/f

### Description
Sets the ATB variant and origin for Aura installs in order to measure retention.

### Steps to test this PR

_Point at the JSON Blob linked in the task_

_Feature enabled and new user Aura install_
- [x] Check out this branch
- [x] Change `installationSourceModern()` in `RealInstallSourceExtractor` to return an Aura package (e.g. `com.sec.android.app.samsungapps`)
- [x] On a fresh emulator, freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is `mq`
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android`
- [x] Verify that `reinstall=false`

_Feature enabled and returning user Aura install_
- [x] On the same emulator, uninstall and freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is `mq`
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android`
- [x] Verify that `reinstall=true`

_Feature disabled and new user Aura install_
- [x] Disable `auraExperiment` in the config
- [x] On a fresh emulator, freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is set normally (Not `mq`)
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android` is **not** sent
- [x] Verify that `reinstall=false`

_Feature disabled and returning user Aura install_
- [x] On the same emulator, uninstall and freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is set normally (Not `mq`)
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android` is **not** sent
- [x] Verify that `reinstall=true`

_Feature enabled and new user install (Not Aura)_
- [x] Re-enable `auraExperiment` in the config
- [x] Change `installationSourceModern()` in `RealInstallSourceExtractor` to return something else (e.g. `com.example`)
- [x] On a fresh emulator, freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is set normally (Not `mq`)
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android` is **not** sent
- [x] Verify that `reinstall=false`

_Feature enabled and returning user install (Not Aura)_
- [x] On the same emulator, uninstall and freshly install the app
- [x] Filter logcat by `Initialized ATB`
- [x] Verify that the variant is set normally (Not `mq`)
- [x] Filter logcat by `m_android_install`
- [x] Verify that `origin=funnel_app_aurapaid_android` is **not** sent
- [x] Verify that `reinstall=true`

### CODE REVIEW

**Comment by joshliebe** [comment]
* **#5343** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5343?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/AtbInitializer.kt [hunk 1]
```diff
@@ -73,8 +75,11 @@ class AtbInitializer @Inject constructor(
 
     override fun onPrivacyConfigDownloaded() {
         if (!statisticsDataStore.hasInstallationStatistics) {
-            // First time we initializeAtb
-            statisticsUpdater.initializeAtb()
+            appCoroutineScope.launch(dispatcherProvider.io()) {
+                auraExperimentManager.initialize()
```
Comment (aitorvs) [inline statistics/statistics-impl/src/main/java/com/duckduckgo/app/statistics/AtbInitializer.kt:79]: I thought we were gonna remove this. Why we still need a public API for this??

**Review by aitorvs — COMMENTED** [review]
I believe we don't need to add code outside the install-impl package. Unless I miss something, everything can just be an implementation detail. No need to add the install-api neither add anything in the :app module

File: app/src/main/java/com/duckduckgo/app/referral/AppReferrerDataStore.kt [hunk 2]
```diff
@@ -29,6 +29,7 @@ interface AppReferrerDataStore {
     var campaignSuffix: String?
     var installedFromEuAuction: Boolean
     var utmOriginAttributeCampaign: String?
+    var returningUser: Boolean
```
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/referral/AppReferrerDataStore.kt:32]: Why are we duplicating `returningUser` storage? now we have two sources of truth

**Review by aitorvs — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/duckduckgo/app/browser/aura/AuraExperimentManagerImpl.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+) : AuraExperimentManager {
+
+    override suspend fun initialize() {
+        if (auraExperimentFeature.self().isEnabled()) {
+            installSourceExtractor.extract()?.let { source ->
+                val settings = auraExperimentFeature.self().getSettings()
+                val packages = auraExperimentListJsonParser.parseJson(settings).list
+                if (packages.contains(source)) {
+                    if (statisticsDataStore.variant == RETURNING_USER) {
+                        appReferrerDataStore.returningUser = true
```
Comment (aitorvs) [inline app/src/main/java/com/duckduckgo/app/browser/aura/AuraExperimentManagerImpl.kt:45]: Said in another place, but why do we need to re-store a returning user flag? it also seems that is used as follows

```kotlin
val returningUser = statisticsDataStore.variant == RETURNING_USER_VARIANT || appReferrerDataStore.returningUser
```
and fori t to be set to true `statisticsDataStore.variant == RETURNING_USER_VARIANT` needs to be `true` so I don't think we need two sources of truth for this

File: app/src/testPlay/java/com/duckduckgo/app/referrer/AppReferrerInstallPixelSenderTest.kt [hunk 4]
```diff
@@ -35,14 +35,16 @@ class AppReferrerInstallPixelSenderTest {
...
     private val appReferrerDataStore: AppReferrerDataStore = mock()
-    private val statisticsDataStore: StatisticsDataStore = mock()
     private val playStoreInstallChecker: VerificationCheckPlayStoreInstall = mock()
     private val captor = argumentCaptor<Map<String, String>>()
 
     @Before
     fun setup() {
         whenever(appBuildConfig.deviceLocale).thenReturn(Locale.US)
         whenever(playStoreInstallChecker.installedFromPlayStore()).thenReturn(true)
+        runBlocking {
```
Comment (CDRussell) [inline app/src/testPlay/java/com/duckduckgo/app/referrer/AppReferrerInstallPixelSenderTest.kt:45]: nit: you can remove `runBlocking` if you define the function as `fun setup() = runTest {  }`

File: statistics/statistics-api/src/main/java/com/duckduckgo/app/statistics/AtbInitializerListener.kt [hunk 5]
```diff
@@ -23,4 +23,9 @@ interface AtbInitializerListener {
 
     /** @return the timeout in milliseconds after which [beforeAtbInit] will be stopped */
     fun beforeAtbInitTimeoutMillis(): Long
+
+    companion object {
+        const val REINSTALL_LISTENER = 10
```
Comment (CDRussell) [inline statistics/statistics-api/src/main/java/com/duckduckgo/app/statistics/AtbInitializerListener.kt:28]: nit: add `PRIORITY` into the key names

File: statistics/statistics-impl/src/test/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculationTest.kt [hunk 6]
```diff
@@ -66,7 +67,7 @@ class SegmentCalculationTest(private val input: TestInput) {
     @Before
     fun setup() {
         atbStore = FakeStatisticsDataStore()
-        whenever(mockVariantManager.getVariantKey()).thenReturn(null)
+        runBlocking { whenever(appBuildConfig.isAppReinstall()).thenReturn(false) }
```
Comment (CDRussell) [inline statistics/statistics-impl/src/test/java/com/duckduckgo/app/statistics/user_segments/SegmentCalculationTest.kt:70]: Nit: As before, can use `fun setup() = runTest {}` instead

**Review by CDRussell — APPROVED** [review]
New approach tested and again, and still working as expected.

File: app/src/main/java/com/duckduckgo/app/buildconfig/RealAppBuildConfig.kt [hunk 7]
```diff
@@ -65,6 +78,65 @@ class RealAppBuildConfig @Inject constructor(
...
+
+            if (preferences.contains(APP_REINSTALLED_KEY)) {
+                return@withContext preferences.getBoolean(APP_REINSTALLED_KEY, false)
+            }
+
+            val isAppReinstalled = preferences.getBoolean(APP_REINSTALLED_KEY, false)
+
+            if (isAppReinstalled) {
+                return@withContext true
+            } else {
```
Comment (nalcalag) [inline app/src/main/java/com/duckduckgo/app/buildconfig/RealAppBuildConfig.kt:95]: **Question:** There’s already a check right before this to see if `APP_REINSTALLED_KEY` has a value, so if I’m not mistaken (though I might be, haha), we could probably remove this part?

Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/buildconfig/RealAppBuildConfig.kt]: @nalcalag Good spot, I’ve removed that check 👍

**Pull request merged**